### PR TITLE
Update tslint.json

### PR DESCRIPTION
**Pull request by PrzemekCraker**
Added new tslint's rules.
To familiarize yourself with the rules check: https://palantir.github.io/tslint/rules/
Usage to entire project: tslint --project tsconfig.json --config tslint.json

### CODE REVIEW

**Review by i0skar — APPROVED** [review]

File: tslint.json [hunk 1]
```diff
@@ -1,14 +1,49 @@
...
+      "property-declaration"
+    ],
+
+    "ordered-imports": true,
+    "object-literal-sort-keys": [
+      true,
+      "ignore-blank-lines",
+      "ignore-case",
+      "locale-compare",
+      "match-declaration-order",
```
Comment (Czaroos) [inline tslint.json:28]: My suggestion is that if we want to stick strictly to alphabetical order (as declared above), we'd rather remove this line.

**Review by Czaroos — COMMENTED** [review]
Check my comment on line 28 and decide whether we should apply this rule or not.

**Review by polubis — APPROVED** [review]
Good job. Could you configure a prettier to do the formatting on these rules?

File: tslint.json [hunk 2]
```diff
@@ -1,14 +1,49 @@
...
-        "no-duplicate-variable": true,
-        "no-unused-variable": true,
-        "triple-equals": true,
-        "prefer-const": true,
-        "arrow-return-shorthand": true,
-        "no-invalid-template-strings": true
-    }
-}
\ No newline at end of file
+  "rules": {
```
Comment (polubis) [inline tslint.json:2]: We need a script that checks package.json also which runs check globally

**Review by polubis — CHANGES_REQUESTED** [review]

File: tslint.json (hunk 1 above)
Comment (PrzemekCraker) [reply tslint.json]: Finally - I deleted "match-declaration-order" cuz was causing an error, thanks for your notice anyway.

File: package.json [hunk 3]
```diff
@@ -9,7 +9,8 @@
     "test": "jest",
     "test-coverage": "jest --coverage",
     "test-watch": "jest --watch",
-    "test-watch-silent": "jest --watch --silent"
+    "test-watch-silent": "jest --watch --silent",
+    "lint": "tslint --project .",
```
Comment (polubis) [inline package.json:13]: Remove comma at end

File: tslint.json [hunk 4]
```diff
@@ -1,14 +1,55 @@
...
-        "no-duplicate-variable": true,
-        "no-unused-variable": true,
-        "triple-equals": true,
-        "prefer-const": true,
-        "arrow-return-shorthand": true,
-        "no-invalid-template-strings": true
-    }
-}
\ No newline at end of file
+  "extends": ["tslint:recommended", "tslint-config-prettier"],
```
Comment (polubis) [inline tslint.json:2]: Find and remove rule which tells about namespace and modules invalid usage - we using namespaces almost in all React files

**Review by polubis — CHANGES_REQUESTED** [review]
Good job. Just 2 small things to fix

File: tslint.json [hunk 5]
```diff
@@ -1,14 +1,55 @@
...
+  "rulesDirectory": ["tslint-plugin-prettier"],
+  "rules": {
+    "prettier": [ true, { "printWidth": 100,
+      "singleQuote": true,
+      "jsxSingleQuote": false,
+      "bracketSpacing": true,
+      "semi": true,
+      "trailingComma": "none",
+      "endOfLine": "crlf" } ],
```
Comment (polubis) [inline tslint.json:11]: Remove prettier.rc file if you using prettier rules inside tslint config

**Review by polubis — CHANGES_REQUESTED** [review]
Good job. Just 2 small things to fix

**Review by polubis — APPROVED** [review]

**Comment by polubis** [comment]
PR approved - but we can't merge it yet. Admin panel must be finished

File: package.json [hunk 6]
```diff
@@ -9,7 +9,8 @@
     "test": "jest",
     "test-coverage": "jest --coverage",
     "test-watch": "jest --watch",
-    "test-watch-silent": "jest --watch --silent"
+    "test-watch-silent": "jest --watch --silent",
+    "lint": "tslint --project ."
```
Comment (polubis) [inline package.json:13]: We need command which fixes lint issues and also command for use prettier
"lint-and-fix": "eslint . --ext .ts --fix",
"prettier-format": "prettier --config .prettierrc **/*.ts --write",

File: tslint.json [hunk 7]
```diff
@@ -1,14 +1,56 @@
...
+    ],
+    "no-duplicate-super": true,
+    "no-duplicate-switch-case": true,
+    "no-shadowed-variable": true,
+    "no-sparse-arrays": true,
+    "no-switch-case-fall-through": true,
+    "no-tautology-expression": true,
+    "variable-name": {
+      "options": ["check-format", "allow-leading-underscore", "allow-pascal-case"]
+    },
```
Comment (polubis) [inline tslint.json:46]: "sort-keys": ["error", "asc", {"caseSensitive": true, "natural": false, "minKeys": 2}] - add sorting rules for react properties and object keys

**Review by polubis — CHANGES_REQUESTED** [review]

**Pull request closed without merging**